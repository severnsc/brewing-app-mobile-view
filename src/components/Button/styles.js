import { StyleSheet } from "react-native";
import { lightGray, gray, primary, success, danger } from "../../constants";

export default StyleSheet.create({
  button: {
    borderColor: lightGray,
    borderWidth: 1,
    alignItems: "center",
    padding: 10,
    backgroundColor: gray
  },
  primary: {
    backgroundColor: primary
  },
  secondary: {
    backgroundColor: "white"
  },
  success: {
    backgroundColor: success
  },
  danger: {
    backgroundColor: danger
  },
  round: {
    borderRadius: 100
  },
  circle: {
    width: 50,
    height: 50,
    borderRadius: 100,
    justifyContent: "center"
  },
  hideBorder: {
    borderWidth: 0
  }
});
