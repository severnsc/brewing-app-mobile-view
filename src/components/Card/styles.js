import { StyleSheet } from "react-native";

export default StyleSheet.create({
	card: {
		padding: 5,
		height: 80,
		justifyContent: "space-between",
		shadowColor: "#000",
		shadowOffset: { width: 4, height: 6 },
		shadowOpacity: 0.2,
		shadowRadius: 2,
		backgroundColor: "white"
	},
	row: {
		flexDirection: "row",
		justifyContent: "space-between"
	}
});
