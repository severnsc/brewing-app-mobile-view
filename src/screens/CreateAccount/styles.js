import { StyleSheet } from "react-native";
import { white } from "../../constants";

export default StyleSheet.create({
	container: {
		justifyContent: "center",
		alignItems: "center",
		backgroundColor: white,
		margin: 5,
		top: 100
	},
	form: {
		width: "100%",
		paddingHorizontal: 20,
		paddingVertical: 10
	},
	input: {
		marginVertical: 10
	}
});
