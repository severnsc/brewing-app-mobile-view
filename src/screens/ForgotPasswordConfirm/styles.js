import { StyleSheet } from "react-native";

export default StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "space-around",
    alignItems: "center",
    padding: 5,
    marginTop: 100
  },
  text: { fontSize: 24, textAlign: "center" }
});
