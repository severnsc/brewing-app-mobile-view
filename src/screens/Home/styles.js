import { StyleSheet } from "react-native";

export default StyleSheet.create({
	container: {
		flex: 1,
		justifyContent: "space-around",
		alignItems: "center",
		padding: 5
	},
	buttonRow: {
		width: "100%"
	},
	button: {
		marginTop: 10
	}
});
