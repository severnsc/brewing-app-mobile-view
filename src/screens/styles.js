import { StyleSheet } from "react-native";
import { white } from "../constants";

export default StyleSheet.create({
  container: {
    alignItems: "center",
    justifyContent: "center",
    top: 100,
    margin: 5,
    backgroundColor: white
  },
  form: {
    width: "100%",
    paddingHorizontal: 20,
    paddingVertical: 10
  },
  input: {
    marginVertical: 10
  }
});
