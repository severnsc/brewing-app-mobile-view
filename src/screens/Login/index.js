import React from "react";
import PropTypes from "prop-types";
import { GradientView, Form, TextInput, Button, Text } from "../../components";
import { KeyboardAvoidingView } from "react-native";
import styles from "./styles";
import { white, primary, INVALID_LOGIN, NETWORK_ERROR } from "../../constants";
import { AlertIOS } from "react-native";

const Login = ({ isError, login, forgotPassword }) => (
  <GradientView>
    <KeyboardAvoidingView style={styles.container}>
      <Form
        onSubmit={login}
        style={styles.form}
        initialValues={[{ id: "1", value: "" }, { id: "2", value: "" }]}
      >
        {(values, onChange, onSubmit) => {
          const username = values[0] && values[0].value;
          const password = values[1] && values[1].value;
          const submit = () => {
            onSubmit().catch(() => AlertIOS.alert("Error!", NETWORK_ERROR));
          };
          console.log(INVALID_LOGIN);
          return (
            <React.Fragment>
              {isError ? (
                <Text
                  danger={true}
                  value={INVALID_LOGIN}
                  testID="loginErrorText"
                />
              ) : null}
              <TextInput
                label="Username"
                onChange={value => onChange("1", value)}
                style={styles.input}
                value={username}
                testID="usernameInput"
              />
              <TextInput
                label="Password"
                password={true}
                onChange={value => onChange("2", value)}
                style={styles.input}
                value={password}
                testID="passwordInput"
              />
              <Button
                value="Login"
                onPress={submit}
                success={true}
                textColor={white}
                testID="submitButton"
              />
              <Button
                value="Forgot password"
                onPress={forgotPassword}
                style={styles.forgotPassword}
                textColor={primary}
              />
            </React.Fragment>
          );
        }}
      </Form>
    </KeyboardAvoidingView>
  </GradientView>
);

Login.propTypes = {
  isError: PropTypes.bool,
  login: PropTypes.func.isRequired,
  forgotPassword: PropTypes.func.isRequired
};

Login.defaultProps = {
  isError: false,
  login: () => {},
  forgotPassword: () => {}
};

export default Login;
