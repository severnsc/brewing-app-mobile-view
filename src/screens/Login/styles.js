import { StyleSheet } from "react-native";
import { white } from "../../constants";
import styles from "../styles";

export default StyleSheet.create({
  ...styles,
  forgotPassword: {
    backgroundColor: white,
    borderColor: white
  }
});
