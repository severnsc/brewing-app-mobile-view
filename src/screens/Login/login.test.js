import React from "react";
import Login from ".";
import { shallow } from "enzyme";
import styles from "./styles";
import { white, primary, INVALID_LOGIN, NETWORK_ERROR } from "../../constants";
import renderer from "react-test-renderer";
import { AlertIOS } from "react-native";

describe("Login screen", () => {
  it("matches snapshot", () => {
    const login = renderer.create(<Login />);
    const tree = login.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("returns a GradientView", () => {
    const login = shallow(<Login />);
    const gradient = login.find("GradientView");
    expect(gradient).toHaveLength(1);
  });
  it("returns a KeyboardAvoidingView with container styles", () => {
    const login = shallow(<Login />);
    const keyboardView = login.find("KeyboardAvoidingView");
    expect(keyboardView).toHaveLength(1);
    expect(keyboardView.prop("style")).toEqual(styles.container);
  });
  it("returns a form with form styles", () => {
    const login = shallow(<Login />);
    const form = login.find("Form");
    expect(form).toHaveLength(1);
    expect(form.prop("style")).toEqual(styles.form);
  });
  describe("form", () => {
    it("has onSubmit prop equal to login prop", () => {
      const login = jest.fn();
      const loginScreen = shallow(<Login login={login} />);
      let form = loginScreen.find("Form");
      expect(form.prop("onSubmit")).toBe(login);
    });
    it("has initialValues prop with two objects with empty strings", () => {
      const login = shallow(<Login />);
      const form = login.find("Form");
      const initialValues = [{ id: "1", value: "" }, { id: "2", value: "" }];
      expect(form.prop("initialValues")).toEqual(initialValues);
    });
    describe("username input", () => {
      const login = shallow(<Login />);
      let form = login.find("Form");
      const username = "username";
      const onChange = jest.fn();
      const onSubmit = jest.fn();
      form = form
        .props()
        .children([{ id: "1", value: username }], onChange, onSubmit);
      const usernameInput = form.props.children[1];
      it("has label Username", () => {
        expect(usernameInput.props.label).toBe("Username");
      });
      it("has style input", () => {
        expect(usernameInput.props.style).toBe(styles.input);
      });
      it("has value equal to value prop of first item on values array", () => {
        expect(usernameInput.props.value).toBe(username);
      });
      it("calls forms provided onChange with 1 and new value onChange", () => {
        usernameInput.props.onChange("newValue");
        expect(onChange).toHaveBeenCalledWith("1", "newValue");
      });
    });
    describe("password input", () => {
      const login = shallow(<Login />);
      let form = login.find("Form");
      const username = "username";
      const password = "password";
      const onChange = jest.fn();
      const onSubmit = jest.fn();
      form = form
        .props()
        .children(
          [{ id: "1", value: username }, { id: "2", value: password }],
          onChange,
          onSubmit
        );
      const passwordInput = form.props.children[2];
      it("has label Password", () => {
        expect(passwordInput.props.label).toBe("Password");
      });
      it("has password prop equal true", () => {
        expect(passwordInput.props.password).toBe(true);
      });
      it("has style input", () => {
        expect(passwordInput.props.style).toBe(styles.input);
      });
      it("has value equal to value prop of the second item returned in the values array", () => {
        expect(passwordInput.props.value).toBe(password);
      });
      it("calls form provided onChange with 2 and new value onChange", () => {
        passwordInput.props.onChange("newPassword");
        expect(onChange).toHaveBeenCalledWith("2", "newPassword");
      });
    });
    describe("login button", () => {
      const login = shallow(<Login />);
      let form = login.find("Form");
      const username = "username";
      const onChange = jest.fn();
      const onSubmit = jest.fn(() => Promise.resolve());
      form = form
        .props()
        .children([{ id: "1", value: username }], onChange, onSubmit);
      const button = form.props.children[3];
      it("has value Login", () => {
        expect(button.props.value).toBe("Login");
      });
      it("has success true", () => {
        expect(button.props.success).toBe(true);
      });
      it("has textColor white", () => {
        expect(button.props.textColor).toBe(white);
      });
      it("calls form provided onSubmit onPress", () => {
        button.props.onPress();
        expect(onSubmit).toHaveBeenCalled();
      });
    });
    describe("forgot password button", () => {
      const forgotPasswordFunc = jest.fn();
      const login = shallow(<Login forgotPassword={forgotPasswordFunc} />);
      let form = login.find("Form");
      const username = "username";
      form = form.props().children([{ id: "1", value: username }]);
      const forgotPassword = form.props.children[4];
      it("has value Forgot password", () => {
        expect(forgotPassword.props.value).toBe("Forgot password");
      });
      it("has forgotPassword style", () => {
        expect(forgotPassword.props.style).toBe(styles.forgotPassword);
      });
      it("has textColor primary", () => {
        expect(forgotPassword.props.textColor).toBe(primary);
      });
      it("calls forgotPassword prop onPress", () => {
        forgotPassword.props.onPress();
        expect(forgotPasswordFunc).toHaveBeenCalled();
      });
    });
    describe("when isError is true", () => {
      it("matches snapshot", () => {
        const login = renderer.create(<Login isError={true} />);
        const tree = login.toJSON();
        expect(tree).toMatchSnapshot();
      });
      it("displays the INVALID_LOGIN error", () => {
        const login = shallow(<Login isError={true} />);
        let form = login.find("Form");
        const username = "username";
        form = form.props().children([{ id: "1", value: username }]);
        const error = form.props.children[0];
        expect(error.type.name).toBe("Text");
        expect(error.props.danger).toBe(true);
        expect(error.props.value).toBe(INVALID_LOGIN);
      });
    });
    describe("submitting", () => {
      it("calls the login prop onSubmit", () => {
        const login = jest.fn();
        const loginScreen = shallow(<Login login={login} />);
        const form = loginScreen.find("Form");
        form.simulate("submit");
        expect(login).toHaveBeenCalled();
      });
      describe("when onSubmit rejects", () => {
        it("displays an AlertIOS with NETWORK_ERROR message", () => {
          const login = jest.fn(() => Promise.reject());
          const onChange = jest.fn();
          const loginScreen = shallow(<Login login={login} />);
          let form = loginScreen.find("Form");
          form = form
            .props()
            .children(
              [{ id: "1", value: "" }, { id: "2", value: "" }],
              onChange,
              login
            );
          const button = form.props.children[3];
          button.props.onPress();
          const spy = jest.spyOn(AlertIOS, "alert");
          return Promise.resolve()
            .then()
            .then(() => {
              expect(spy).toHaveBeenCalledWith("Error!", NETWORK_ERROR);
            });
        });
      });
    });
  });
});
