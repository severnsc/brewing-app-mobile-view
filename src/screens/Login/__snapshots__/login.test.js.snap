// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login screen form login button when loading value is true matches snapshot 1`] = `
<Component
  animating={true}
  color="#999999"
  hidesWhenStopped={true}
  size="small"
/>
`;

exports[`Login screen form when isError is true matches snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "justifyContent": "center",
      "margin": 5,
      "top": 100,
    }
  }
>
  <View
    style={
      Object {
        "paddingHorizontal": 20,
        "paddingVertical": 10,
        "width": "100%",
      }
    }
  >
    <Text
      accessibilityLabel="Invalid Username or Password!"
      style={
        Object {
          "color": "#dc302f",
        }
      }
      testID="loginErrorText"
    >
      Invalid Username or Password!
    </Text>
    <View
      style={
        Object {
          "borderBottomColor": undefined,
          "borderBottomWidth": 1,
          "marginVertical": 10,
        }
      }
    >
      <Text
        accessibilityLabel="Username"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        Username
      </Text>
      <TextInput
        autoFocus={false}
        keyboardType="default"
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        testID="usernameInput"
        value=""
      />
    </View>
    <View
      style={
        Object {
          "borderBottomColor": undefined,
          "borderBottomWidth": 1,
          "marginVertical": 10,
        }
      }
    >
      <Text
        accessibilityLabel="Password"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        Password
      </Text>
      <TextInput
        autoFocus={false}
        keyboardType="default"
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        secureTextEntry={true}
        testID="passwordInput"
        value=""
      />
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#37c172",
          "borderColor": "#e9ecee",
          "borderWidth": 1,
          "opacity": 1,
          "padding": 10,
        }
      }
      testID="submitButton"
    >
      <Text
        accessibilityLabel="Login"
        style={
          Object {
            "color": "#fff",
            "fontSize": undefined,
          }
        }
      >
        Login
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "#fff",
          "borderWidth": 1,
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <Text
        accessibilityLabel="Forgot password"
        style={
          Object {
            "color": "#3183c8",
            "fontSize": undefined,
          }
        }
      >
        Forgot password
      </Text>
    </View>
  </View>
</View>
`;

exports[`Login screen matches snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "justifyContent": "center",
      "margin": 5,
      "top": 100,
    }
  }
>
  <View
    style={
      Object {
        "paddingHorizontal": 20,
        "paddingVertical": 10,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "borderBottomColor": undefined,
          "borderBottomWidth": 1,
          "marginVertical": 10,
        }
      }
    >
      <Text
        accessibilityLabel="Username"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        Username
      </Text>
      <TextInput
        autoFocus={false}
        keyboardType="default"
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        testID="usernameInput"
        value=""
      />
    </View>
    <View
      style={
        Object {
          "borderBottomColor": undefined,
          "borderBottomWidth": 1,
          "marginVertical": 10,
        }
      }
    >
      <Text
        accessibilityLabel="Password"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        Password
      </Text>
      <TextInput
        autoFocus={false}
        keyboardType="default"
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        secureTextEntry={true}
        testID="passwordInput"
        value=""
      />
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#37c172",
          "borderColor": "#e9ecee",
          "borderWidth": 1,
          "opacity": 1,
          "padding": 10,
        }
      }
      testID="submitButton"
    >
      <Text
        accessibilityLabel="Login"
        style={
          Object {
            "color": "#fff",
            "fontSize": undefined,
          }
        }
      >
        Login
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "#fff",
          "borderWidth": 1,
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <Text
        accessibilityLabel="Forgot password"
        style={
          Object {
            "color": "#3183c8",
            "fontSize": undefined,
          }
        }
      >
        Forgot password
      </Text>
    </View>
  </View>
</View>
`;
